reference/conanfile/methods/configure.rst

```diff
@@ -0,0 +1,60 @@
+.. _reference_conanfile_methods_configure:
+
+configure()
+===========
+
+The ``configure()`` method should be used for configuration of settings and options in the recipe
+for later use in the different methods like ``generate()``, ``build()`` or ``package()``. This
+method executes while building the dependency graph and expanding the packages dependencies, that means
+that when this method executes the dependencies are still not there, they do not exist, and it is not
+possible to access ``self.dependencies``.
+
+For example, for a C (not C++) library, the ``compiler.libcxx`` and ``compiler.cppstd`` shouldn't
+even exist during the ``build()``. It is not only that they are not part of the ``package_id``, but
+they shouldn't be used in the build process at all. They will be defined in the profile, because
+other packages in the graph can be C++ packages and need them, but it is the responsibility of this
+recipe to remove them so they are not used in the recipe:
+
+.. code-block:: python
+    
+    settings = "os", "compiler", "build_type", "arch"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
```

Comment:
perhaps, should use `rm_safe`?
https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#settings

Reply:
Yes, good catch, thankyou, changing it.